diff --git a/psiphon/common/tapdance/tapdance.go b/psiphon/common/tapdance/tapdance.go
index de08e792..0b857300 100644
--- a/psiphon/common/tapdance/tapdance.go
+++ b/psiphon/common/tapdance/tapdance.go
@@ -327,8 +327,19 @@ func Dial(
 
 	manager := newDialManager(netDialer.DialContext)
 
+	registrar := refraction_networking_tapdance.APIRegistrar{
+		Endpoint:           "https://registration.refraction.network",
+		ConnectionDelay:    250 * time.Millisecond,
+		MaxRetries:         3,
+		SecondaryRegistrar: refraction_networking_tapdance.DecoyRegistrar{},
+	}
+
 	tapdanceDialer := &refraction_networking_tapdance.Dialer{
-		TcpDialer: manager.dial,
+		TcpDialer:          manager.dial,
+		DarkDecoy:          true,
+		UseProxyHeader:     true,
+		Width:              5,
+		DarkDecoyRegistrar: registrar,
 	}
 
 	// If the dial context is cancelled, use dialManager to interrupt
